vic/drivers/image/src/vic_image_start.c

```diff
@@ -45,6 +45,17 @@ vic_image_start(void)
         get_global_param(filep.globalparam);
     }
 
+    if (mpi_rank == VIC_MPI_ROOT) {
+        // Open domain and parameter netCDF files
+        filenames.domain.nc_id = open_nc(filenames.domain.nc_file);
+        filenames.params.nc_id = open_nc(filenames.params.nc_file);
+    }
+
     // initialize image mode structures and settings
     vic_start();
+
+    if (mpi_rank == VIC_MPI_ROOT) {
+        // Close domain file
+        close_nc(filenames.domain);
+    }
```

Comment:
Basically if the information is not needed at a higher level in the code then it is better to hide it (put it "lower" in the code) and put the code where it is used. We want to keep the high level functions as clean as possible. Also keep in mind that opening the parameter file once or twice is not all that expensive. It's expensive when we have to open it for every read.